include/fuse.h

```diff
@@ -574,6 +574,22 @@ struct fuse_operations {
 	 */
 	int (*fallocate) (const char *, int, off_t, off_t,
 			  struct fuse_file_info *);
+
+	/**
+	 * Change the permission bits of a file
+	 *
+	 * This method is called instead of the chmod() method if the
+	 * call was invoked from an fchmod() system call.
+	 *
+	 * TODO: Determine the version of the Linux kernel below.
+	 *
+	 * If this method is not implemented or under Linux kernel
+	 * versions earlier than X.X.X, the chmod() method will be
+	 * called instead.
```

Comment:
I don't see any dependence on the Linux kernel anywhere. You've changed how FUSE responds to the setattr request. Why / how do you think the Linux kernel version would come into effect?